Modules/Tests/NetworkingTests/ApplicationPassword/RequestProcessorTests.swift

```diff
@@ -445,6 +445,51 @@ final class RequestProcessorTests: XCTestCase {
         XCTAssertTrue(mockRequestAuthenticator.deleteApplicationPasswordCalled)
         XCTAssertFalse(shouldRetry.retryRequired)
     }
+
+    /// The test is designed for manual local run. It must be excluded from CI pipelines.
+    /// Simulates race condition of multiple `updateAuthenticator` calls from different threads
+    /// Should cause a crash on old/non-synchronized `RequestProcessor` implementatoin
+    /// Should work fine on synchronized version of `RequestProcessor`
+    func test_concurrentAuthenticatorUpdatesTriggersRaceWhenUnsynchronized() throws {
```

Comment:
Nit: With the current fix in place, the unit test should pass (no race condition happens). So I think this test should be renamed and we can safely run it on CI. Should there's any crash, that would mean the fix does not work as we expect. WDYT? 


```suggestion
    func test_concurrent_authenticator_updates_does_not_trigger_race_condition() throws {
```